### PR TITLE
ObjectiveC: support cateogries

### LINKED ISSUE #1487: Objective C: inherits field is not impelemented, and categories are not captured.

### PR DESCRIPTION
**Pull request by masatake**
About category of Objective-C, see https://medium.com/ios-os-x-development/an-introduction-to-categories-in-objective-c-d6d55d91569f

Close #1487

### CODE REVIEW

File: parsers/objc.c [hunk 1]
```diff
@@ -126,6 +131,24 @@ static const keywordTable objcKeywordTable[] = {
...
 
+typedef enum {
+	F_CATEGORY,
+	F_PARTOF,
+} objcField;
+
+static fieldDefinition ObjcFields [] = {
+	{
+		.name = "category",
+		.description = "category attached to th class",
```
Comment (b4n) [inline parsers/objc.c]: *to th**e** class* (e is missing)

Comment (masatake) [reply parsers/objc.c]: Thank you. I will update in soon.

**Comment by masatake** [comment]
After thinking again, defining two kinds, categoryIface and categoryImpl looks overkill.
I will revise this.

**Comment by masatake** [comment]
Updated.
categoryImpl and categoryIface kinds are removed. Instead, I introduce category kind.
We can known whether a category is defined in @interface  line or @implementation line from scope information. So we don't have to introduce the two kinds.
Using scope field to know whether a category is defined in @interface  line or @implementation line is
a bit abuse. However, I think it is better than introducing the two kinds.

In addition, I introduced code for parsing protocols spec in @interface line.

**Pull request merged**